src/main/java/service/DriverSetup.java

```diff
@@ -0,0 +1,22 @@
+package service;
+
+import io.appium.java_client.android.AndroidDriver;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class DriverSetup {
+    protected static AndroidDriver driver;
+
+    protected static void prepareDriverNative() throws MalformedURLException {
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+        capabilities.setCapability("deviceName", "emulator-5554");
+        capabilities.setCapability("platformName", "Android");
+        File appDir = new File("C:\\Users\\Volodymyr_Protsiv\\Desktop\\JavaMentoringProgram\\M15_Automated Testing of Mobile Applications with Java\\mobile-testing\\src\\main\\resources");

```

Comment:
Hardcoded values?

Reply:
fixed

Reply:
fixed

Reply:
Create private static String APPLICATION_PATH
Remove line 17
In line 18 use your created APPLICATION_PATH variable except appDir